70_FLIPFluids/app_resources/common.hlsl

```diff
@@ -0,0 +1,38 @@
+#ifndef _FLIP_EXAMPLE_COMMON_HLSL
+#define _FLIP_EXAMPLE_COMMON_HLSL
+
+#include "nbl/builtin/hlsl/cpp_compat.hlsl"
+
+#define NUM_THREADS 128
+
+NBL_CONSTEXPR uint32_t WorkgroupSize = NUM_THREADS;
+NBL_CONSTEXPR float ratioFLIPPIC = 0.95;
+NBL_CONSTEXPR float deltaTime = 1.0f / 60.0f;
+NBL_CONSTEXPR float gravity = 15.0f;
+
+#ifdef __HLSL_VERSION
+
+static const float FLT_MIN = 1.175494351e-38;
+static const float FLT_MAX = 3.402823466e+38;
+
+struct Particle
+{
+    float4 position;
+    float4 velocity;
+
+    uint id;
+    uint pad[3];
+};
```

Comment:
1. we always compile with `fvk-use-scalar-layout` so you should raraley ever put any padding (only in cases C++ would emit it)
2. [don't change this now] use sized ints and floats, especially the ints so `uint32_t` and `int32_t`

Reply:
What's even the need to keep an `id` in the particle?

Also, because only certain shaders/dispatches only access certain properties of a particle, not everyone needs to know the `velocity`.

Therefore I'd use "Structure of Arrays" rather than "Array of Structures"
```cpp
struct Particles
{
   nbl::hlsl::BdaAccessor<float32_t3> positions;
   nbl::hlsl::BdaAccessor<float32_t3> velocities;
};
```
with BDA of course